include/LightGBM/objective_function.h

```diff
@@ -566,6 +566,9 @@ class ConstrainedObjectiveFunction : public ObjectiveFunction
         // Update index of multipliers to be used for next constraints
         multipliers_base_index += 1;
       }
+
+      gradients[i] *= num_data_;
```

Comment:
This multiplication has to be done in each if clause, for each loss function, instead of at the end.

The `gradients` array is not zeroed out at the beginning of this function, it already has gradients from the predictive loss (and we only want to scale gradients from constraint loss).